server/src/main/java/org/elasticsearch/index/engine/SoftDeletesPolicy.java

```diff
@@ -102,9 +102,17 @@ synchronized long getMinRetainedSeqNo() {
             // then sends ops after the local checkpoint of that commit. This requires keeping all ops after localCheckpointOfSafeCommit;
             // - Changes APIs are driven the combination of the global checkpoint and retention ops. Here we prefer using the global
             // checkpoint instead of max_seqno because only operations up to the global checkpoint are exposed in the the changes APIs.
-            final long minSeqNoForQueryingChanges = globalCheckpointSupplier.getAsLong() - retentionOperations;
+            final long retentionOps;
+            if (retentionSizeInBytes > 0) {
+                // Use 1 byte per document if the average doc size is not available yet.
```

Comment:
can you state when that is the case, when do we not have an avgDocSize? and in this case should we keep unlimited instead?